src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -80,6 +83,9 @@ export class TerminalService implements ITerminalService {
 			this.setActiveInstanceByIndex(0);
 		}
 		this._onInstancesChanged.fire();
+		this._updateCursorBlink(terminalInstance);
```

Comment:
Absolutely correct, piggybacking on `TerminalInstance.attachToElement` makes a lot of sense. That part of the config is now no longer a concern of `TerminalPanel`, the `TerminalInstance` is only responsible for itself and `TerminalService` handles the global updates when settings change. Seems reasonable 👍